qutebrowser/browser/commands.py

```diff
@@ -1877,3 +1884,46 @@ def edit_url(self, url=None, bg=False, tab=False, window=False,
             self.openurl, bg=bg, tab=tab, window=window, count=count))
 
         ed.edit(url or self._current_url().toString())
+
+    @cmdutils.register(instance='command-dispatcher', scope='window')
+    def set_mark(self, key):
+        """Set a mark at the current scroll position in the current tab.
+
+        Args:
+            key: mark identifier; capital indicates a global mark
+        """
+        # consider urls that differ only in fragment to be identical
+        self._tabbed_browser.set_mark(key)
+
+    @cmdutils.register(instance='command-dispatcher', scope='window')
+    def jump_mark(self, key):
+        """Jump to the mark named by `key`.
+
+        Args:
+            key: mark identifier; capital indicates a global mark
+        """
+        try:
+            y, url = self._tabbed_browser.get_mark(key)
+        except MarkNotSetError as e:
+            message.error(self._win_id, str(e))
+            return
+
+        if url is None:
+            # save the pre-jump position in the special ' mark
+            # this has to happen after we read the mark, otherwise jump_mark
+            # "'" would just jump to the current position every time
+            self.set_mark("'")
+            self._scroll_px_absolute(y)
+        else:
+            def callback(ok):
+                if ok:
+                    self._tabbed_browser.cur_load_finished.disconnect(callback)
+                    self._scroll_px_absolute(y)
+
+            self.openurl(url.toString())
+            self._tabbed_browser.cur_load_finished.connect(callback)
+
+    def _scroll_px_absolute(self, y):
+        """Scroll to the position y pixels from the top of the page."""
+        self.scroll('top')
+        self.scroll_px(0, y)
```

Comment:
You can do `self._current_widget().page().currentFrame().setScrollPosition(QPoint(0, y))` instead to do this in one step.